### PR TITLE
Added setreg for UI viewport and animation viewport to limit feature …

### PR DESCRIPTION
**Pull request by VickyAtAZ**
…processors enabled for their scenes.

https://github.com/o3de/o3de/issues/10931

Signed-off-by: Qing Tao <55564570+VickyAtAZ@users.noreply.github.com>

## What does this PR do?

The PR added setreg files for UI viewport and Animation viewport to limit feature processors enabled when create their scene. This will reduce memory overhead and also avoid crash issue reported in https://github.com/o3de/o3de/issues/10931.
This also address part of the issue mentioned in https://github.com/o3de/o3de/issues/6272

## How was this PR tested?

Open the TerrainPhysicsCollider_MaterialMapping_Works.prefab level in o3de Editor. 
Open UI editor then open the lyshine sample uicanvas file. 
Open Animation editor then open a actor file.

### CODE REVIEW

**Review by gadams3 — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Registry/AnimViewport.setreg [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+							// These FPs can be removed once the LightCullingPass issue [ATOM-17812] is addressed 
+							"AZ::Render::SimplePointLightFeatureProcessor",
+							"AZ::Render::SimpleSpotLightFeatureProcessor",
+							"AZ::Render::PointLightFeatureProcessor",
+							"AZ::Render::DiskLightFeatureProcessor",
+							"AZ::Render::CapsuleLightFeatureProcessor",
+							"AZ::Render::QuadLightFeatureProcessor",
+							"AZ::Render::DecalTextureArrayFeatureProcessor"
+						]
+					}
```
Comment (invertednormal) [inline Gems/AtomLyIntegration/EMotionFXAtom/Registry/AnimViewport.setreg:26]: fix tabs

File: Gems/LyShine/Registry/UiViewport.setreg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+							// These FPs can be removed once the LightCullingPass issue [ATOM-17812] is addressed 
+							"AZ::Render::SimplePointLightFeatureProcessor",
+							"AZ::Render::SimpleSpotLightFeatureProcessor",
+							"AZ::Render::PointLightFeatureProcessor",
+							"AZ::Render::DiskLightFeatureProcessor",
+							"AZ::Render::CapsuleLightFeatureProcessor",
+							"AZ::Render::QuadLightFeatureProcessor",
+							"AZ::Render::DecalTextureArrayFeatureProcessor"
+						]
+					}
```
Comment (invertednormal) [inline Gems/LyShine/Registry/UiViewport.setreg:18]: fix tabs

**Review by invertednormal — APPROVED** [review]
looks good, and looks like it'll merge perfectly with my fix too 😄

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp [hunk 3]
```diff
@@ -56,16 +56,18 @@ namespace EMStudio
...
         m_scene = AZ::RPI::Scene::CreateScene(sceneDesc);
-        m_scene->EnableAllFeatureProcessors();
-        // Disable the terrain feature processor as we don't need terrian in anim editor.
-        const AZ::Name terrainFeatureProcessor("TerrainFeatureProcessor");
-        if (m_scene->GetFeatureProcessor(terrainFeatureProcessor))
+
+        if (!sceneDescLoaded)
         {
-            m_scene->DisableFeatureProcessor(terrainFeatureProcessor);
+            AZ_Warning("AnimViewportRenderer", false, "Unable to load setting registery for the viewport's scene settings. Enable all feature processors.");
```
Comment (mbalfour-amzn) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp:69]: Nit: "setting registery" -> "settings registry".
Suggestion: Reword to make it clear how to fix the issue, something like - ("Settings registry is missing the scene settings for this viewport, so all feature processors will be enabled. To enable only a minimal set, add the specific list of feature processors with a registry path of '%s'.", viewportSettingsPath)

File: Gems/LyShine/Code/Source/UiRenderer.cpp [hunk 4]
```diff
@@ -93,11 +94,19 @@ void UiRenderer::OnBootstrapSceneReady(AZ::RPI::Scene* bootstrapScene)
...
     sceneDesc.m_nameId = AZ::Name("UiRenderer");
+    auto settingsRegistry = AZ::SettingsRegistry::Get();
+    const char* viewportSettingPath = "/O3DE/Editor/Viewport/UI/Scene";
+    bool sceneDescLoaded = settingsRegistry->GetObject(sceneDesc, viewportSettingPath);
     AZ::RPI::ScenePtr atomScene = AZ::RPI::Scene::CreateScene(sceneDesc);
-    atomScene->EnableAllFeatureProcessors(); // [LYSHINE_ATOM_TODO][GHI #6272] Enable minimal feature processors
+
+    if (!sceneDescLoaded)
+    {
+        //AZ_Warning("UiRenderer", false, "Unable to load setting registery for the viewport's scene settings. Enable all feature processors.");
```
Comment (mbalfour-amzn) [inline Gems/LyShine/Code/Source/UiRenderer.cpp:107]: Seems like this warning should be enabled? Also, per above, I'd suggest rewording to explain how to fix the problem.

**Review by mbalfour-amzn — APPROVED** [review]

**Review by amzn-rhhong — APPROVED** [review]

File: Gems/LyShine/Code/Source/UiRenderer.cpp (hunk 4 above)
Comment (VickyAtAZ) [reply Gems/LyShine/Code/Source/UiRenderer.cpp:107]: that was a mistake. it should be enabled.
